workers/unity/Assets/Playground/Editor/SnapshotGenerator/SnapshotGenerator.cs

```diff
@@ -16,112 +14,99 @@ internal static class SnapshotGenerator
         public struct Arguments
         {
             public int NumberEntities;
-            public float WorldLength;
             public string OutputPath;
         }
 
-        private const string UnityGameLogicType = "UnityGameLogic";
-        private const string UnityClientType = "UnityClient";
-
-        private static readonly WorkerRequirementSet AllWorkersSet = new WorkerRequirementSet(
-            new Improbable.Collections.List<WorkerAttributeSet>
-            {
-                new WorkerAttributeSet(new Improbable.Collections.List<string> { UnityGameLogicType }),
-                new WorkerAttributeSet(new Improbable.Collections.List<string> { UnityClientType })
-            });
-
-        private static readonly WorkerRequirementSet WorkerSet = new WorkerRequirementSet(
-            new Improbable.Collections.List<WorkerAttributeSet>
-            {
-                new WorkerAttributeSet(new Improbable.Collections.List<string> { UnityGameLogicType })
-            });
-
-        private const string EntityName = "Cube";
+        private static readonly List<string> UnityWorkers =
+            new List<string> { UnityGameLogic.WorkerType, UnityClient.WorkerType };
 
         public static void Generate(Arguments arguments)
         {
             Debug.Log("Generating snapshot.");
-            var entities = CreateEntities(arguments.WorldLength, arguments.NumberEntities);
-            WriteSnapshot(entities, arguments.OutputPath);
+            var snapshot = CreateSnapshot(arguments.NumberEntities);
+
+            Debug.Log($"Writing snapshot to: {arguments.OutputPath}");
+            snapshot.Serialize(arguments.OutputPath);
```

Comment:
Instead of serialize - what about `WriteOut`? Given that strictly all the serialisation is done before that function call.

Reply:
`WriteToDisk`